### ISSUE TITLE
Performance problems with dynamic rendered blocks.

### ISSUE THREAD

**Issue by grogou** [woothemes/woocommerce#42647]
## Describe the bug

I don't know is it bug or no, but the problem is on frontend.
Is there all oki with this js files loaded on frontend? 

![image](https://user-images.githubusercontent.com/10712881/156740366-9a64506b-d6a4-42e4-94d2-60d6b59e8631.png)

Because for example block-editor script loaded on frontend.

![image](https://user-images.githubusercontent.com/10712881/156740603-402de696-5dcb-4390-99c7-bdd2d7338999.png)

Is there a way to fix this? because google page speed drop performance metrics.
For example cart block, the only block in the page and we have metrics like this.

![image](https://user-images.githubusercontent.com/10712881/156741358-a9f65742-b502-4936-b281-0c3d3233dc3e.png)

**Comment by sunyatasattva** [comment]
Thanks for the report, @grogou.

I'll take this as a nice start to talk about this with the team. It's probably going to take time before we address this as we need some big architecture decisions to be made, but I'll keep this issue as a reference for now.

**Comment by grogou** [comment]
@sunyatasattva thank you for your response, after some checks I can see the problem is @wordpress/element render function.
When it is called on front js the whole wordpress packages loaded on frontend for no reason. So maybe the problem is not woocommerce blocks, maybe it is wordpress core or gutenberg plugin issue, or element package issue. It will be nice to solve this, because the only thing in my opinion blocks need to solve the performance issue, if this will be fixed cart and checkout will be incredible blocks to use on stores, I already use it in some projects and script defer not help that much, anyway unused scripts drop metrics in google page speed.

**Comment by sunyatasattva** [comment]
This is an excellent point and thanks for the positive feedback. We are trying our best, I'll keep you posted!

**Comment by grogou** [comment]
@sunyatasattva thank you too.

**Comment by grogou** [comment]
This ticket is no longer actual?

**Comment by sunyatasattva** [comment]
@grogou It still is, as a matter of fact. We have a bot which tries to clean up our increasing backlog. I'll remove the `stale` label and reopen it.

**Comment by grogou** [comment]
@sunyatasattva  thank you)

**Comment by grogou** [comment]
Maybe we can close this? xd

**Comment by sunyatasattva** [comment]
Is this not still the case, though? I believe performance should be somewhat important to us and perhaps we should bump the prio of this instead.

**Comment by grogou** [comment]
Yes I am about that sorry. It is 2 times already on stale status so. If I can help with that I will. Thank you.

**Comment by gigitux** [comment]
It's great to see the significant performance improvements achieved in recent years:  

![Image](https://github.com/user-attachments/assets/45cbf17a-285b-484a-b7aa-868fefc31f8b)  

*(Benchmark performed on the `/shop` page)*  

I believe it's appropriate to close this issue, but at the same time I want to highlight that we are continuing to work to improve the performance.
